### PR TITLE
Blog - Rust Concurrency Patterns for Parallel Programming

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks

### CODE REVIEW

**Comment by msali123** [comment]
@jalletto [Preview Link](https://64aa73c751861f093b3a2407--ephemeral-narwhal-3ac3b2.netlify.app/blog/rust-concurrency-patterns-parallel-programming/)

**Comment by msali123** [comment]
@jalletto [Preview Link](https://64aa7b8f3571ea0b96653001--ephemeral-narwhal-3ac3b2.netlify.app/blog/rust-concurrency-patterns-parallel-programming/)

File: blog/_posts/2023-07-20-rust-concurrency-patterns-parallel-programming.md [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+internal-links:
+ - patterns for parallel programming
+ - rust concurrency patterns
+ - concurrency patterns
+ - parallel programming
+---
+
+**We're [Earthly.dev](https://earthly.dev/). We make building software simpler and, therefore faster – like Dockerfile and Makefile had a baby. This is about Rust Concurrency Patterns for Parallel Programming**
+
+[Rust](https://www.rust-lang.org/) is a modern programming language that prioritizes performance, safety, and concurrency. It's a unique language thanks to its memory safety guarantees, [ownership](https://doc.rust-lang.org/book/ch04-01-what-is-ownership.html#what-is-ownership), and [borrowing](https://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html#references-and-borrowing) system, and support for [fearless concurrency](https://doc.rust-lang.org/book/ch16-00-concurrency.html).
```
Comment (jalletto) [inline blog/_posts/2023-07-20-rust-concurrency-patterns-parallel-programming.md:18]:
```suggestion

Rust is a modern programming language that prioritizes performance, safety, and concurrency. It's a unique language thanks to its memory safety guarantees, ownership, and borrowing system, and support for [fearless concurrency](https://doc.rust-lang.org/book/ch16-00-concurrency.html).
```

File: blog/_posts/2023-07-20-rust-concurrency-patterns-parallel-programming.md [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+
+One of the most basic primitives for concurrency in Rust is threads. A thread is an independent path of execution within a program that can run concurrently with other threads.
+
+Threads allow developers to take full advantage of multicore processors by dividing a task into smaller subtasks that can be executed in parallel.
+
+Now that you know what threads are, let's explore how threads are created and joined in Rust.
+
+### Creating and Joining Threads
+
+To create a new thread in Rust, you can use the [`std::thread::spawn`](https://doc.rust-lang.org/std/thread/fn.spawn.html) function, which requires a [closure](https://doc.rust-lang.org/book/ch13-01-closures.html) as its argument. This closure contains the code that executes in the new thread.
```
Comment (jalletto) [inline blog/_posts/2023-07-20-rust-concurrency-patterns-parallel-programming.md:40]:
```suggestion

To create a new thread in Rust, you can use the [`std::thread::spawn`](https://doc.rust-lang.org/std/thread/fn.spawn.html) function, which requires a closure as its argument. This closure contains the code that executes in the new thread.
```

**Review by jalletto — APPROVED** [review]

**Pull request merged**